File: classes/parte-5-funciones-y-modularidad/074-parametros-por-defecto-y-opcionales/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

// Go no tiene defectos: se simula con la lógica de llamada.
func potencia(base int64, exp int) int64 {
	var r int64 = 1
	for i := 0; i < exp; i++ {
		r *= base
	}
	return r
}

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	t := strings.Fields(line)
	base, _ := strconv.ParseInt(t[0], 10, 64)
	exp := 2
	if len(t) > 1 {
		exp, _ = strconv.Atoi(t[1])
	}
	fmt.Printf("resultado=%d\n", potencia(base, exp))
}

File: classes/parte-5-funciones-y-modularidad/074-parametros-por-defecto-y-opcionales/implementaciones/c/main.c
#include <stdio.h>

/* C no tiene defectos: se simula pasando siempre el exponente. */
long potencia(long base, int exp) {
    long r = 1;
    for (int i = 0; i < exp; i++) r *= base;
    return r;
}

int main(void) {
    long base;
    int exp;
    int leidos = scanf("%ld %d", &base, &exp);
    if (leidos < 1) return 1;
    if (leidos < 2) exp = 2;
    printf("resultado=%ld\n", potencia(base, exp));
    return 0;
}
